integrations/jira/jira/client.py

```diff
@@ -103,107 +176,95 @@ async def create_events_webhook(self, app_host: str) -> None:
-        params = self._generate_base_req_params()
-
-        total_users = len(await self._get_users_data(params))
-
-        if total_users == 0:
-            logger.warning(
-                "User query returned 0 users, did you provide the correct Jira API credentials?"
-            )
-
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] < total_users:
-            logger.info(f"Current query position: {params['startAt']}/{total_users}")
-
-            user_response_list = await self._get_users_data(params)
-
-            if not user_response_list:
-                logger.warning(f"No users found at {params['startAt']}")
-                break
-
-            logger.info(
-                f"Retrieved users: {len(user_response_list)} "
-                f"(Position: {params['startAt']}/{total_users})"
-            )
-
-            yield user_response_list
-            params["startAt"] += PAGE_SIZE
-
-    async def get_single_user(self, account_id: str) -> dict[str, Any]:
-        user_response = await self.client.get(
-            f"{self.api_url}/user", params={"accountId": account_id}
-        )
-        user_response.raise_for_status()
-        return user_response.json()
+        async for users in self._get_paginated_data(f"{self.api_url}/users/search"):
+            yield users
+
+    async def get_paginated_teams(
+        self, org_id: str
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        logger.info("Getting teams from Jira")
+
+        base_url = f"{self.jira_url}/gateway/api/public/teams/v1/org/{org_id}/teams"
+
+        async for teams in self._get_cursor_paginated_data(
+            url=base_url, method="GET", extract_key="entities", cursor_param="cursor"
+        ):
+            yield teams
+
+    async def get_paginated_team_members(
+        self, team_id: str, org_id: str, page_size: int = PAGE_SIZE
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        logger.info(f"Getting members for team {team_id}")
+        url = f"{self.teams_base_url}/org/{org_id}/team/{team_id}/members"
+
+        async for members in self._get_cursor_paginated_data(
+            url,
+            method="POST",
+            extract_key="results",
+            initial_params={"first": page_size},
+            cursor_param="after",
+        ):
+            logger.info(f"Retrieved {len(members)} members for team {team_id}")
+            yield members
+
+    async def enrich_teams_with_members(
+        self, teams: List[Dict[str, Any]], org_id: str
+    ) -> List[Dict[str, Any]]:
+        logger.info(f"Enriching {len(teams)} teams with member information")
+
+        enriched_teams = []
+        for team in teams:
+            team_id = team["teamId"]
+            team_members = []
+
+            async for members_batch in self.get_paginated_team_members(team_id, org_id):
+                team_members.extend(members_batch)
+
+            team["__members"] = team_members
+            enriched_teams.append(team)
+
```

Comment:
instead of processing this sequentially, can we do it in parallel using async.io gather?